torchaudio/datasets/librispeech.py

```diff
@@ -125,9 +125,7 @@ def __init__(self,
                     download_url(url, root, hash_value=checksum)
                 extract_archive(archive)
 
-        walker = walk_files(
-            self._path, suffix=self._ext_audio, prefix=False, remove_suffix=True
-        )
+        walker = sorted([str(p.stem) for p in Path(self._path).glob('*/*/*' + self._ext_audio)])
         self._walker = list(walker)
```

Comment:
Could you get rid of `list(walker)` as it is redundant?

I would do

```python
self._walker =  [str(p.stem) for p in Path(self._path).glob(f'*/*/*{self._ext_audio}')]
self._walker.sort()
```

Reply:
Good point. It seems like using iterator+sorted may be faster? Is this a fair comparison?

```
In [7]: %timeit sorted(range(1000, 0, -1))                                                                                                                                                                                                          
14.3 µs ± 29.6 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [8]: def func(): 
   ...:     a = list(range(1000, 0, -1)) 
   ...:     a.sort() 
   ...:                                                                                                                                                                                                                                             

In [9]: %timeit func()                                                                                                                                                                                                                              
15 µs ± 201 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

Reply:
I've updated the PR with sorted(generator). Once the tests pass, I'll merge this PR as it is. Thanks @krishnakalyan3 for the quick follow-up :)

Reply:
List comprehension is not a generator, please revert it.

Reply:
ah, okay, you changed the list comprehension to a generator.

Reply:
The next time, please consult it with me before doing so.

Reply:
I cannot find a reference that says `sorted` is in-place. can you give the reference for that?

Reply:
See [implementation of sorted](https://github.com/python/cpython/blob/226a012d1cd61f42ecd3056c554922f359a1a35d/Python/bltinmodule.c#L2218). Regardless, the [test](https://github.com/pytorch/audio/pull/1069#discussion_r542804733) above points to that implementation to be faster anyway.